ini/stm32f4.ini

```diff
@@ -166,10 +166,28 @@ extra_scripts     = ${stm32_variant.extra_scripts}
 extends           = stm32_variant
 board             = marlin_opulo_lumen_rev4
 build_flags       = ${stm32_variant.build_flags}
-  -DARDUINO_BLACK_F407VE
-  -DUSBD_USE_CDC_COMPOSITE -DUSE_USB_FS -DHAS_STM32_UID
+                    -DARDUINO_BLACK_F407VE
+                    -DUSBD_USE_CDC_COMPOSITE -DUSE_USB_FS -DHAS_STM32_UID
 extra_scripts     = ${stm32_variant.extra_scripts}
 
+#
+# Langgo407 (STM32F407VE)
+#
+[env:LANGGO407VE]
+extends                    = stm32_variant
+platform                   = ${common_stm32.platform}
+board                      = marlin_STM32F407VE_LANGGO
+board_build.offset         = 0x8000
+board_build.offset_address = 0x08008000
+build_flags                = ${common_stm32.build_flags}
+                             -DARDUINO_BLACK_F407ZE
+                             -DST32_SHIP
+                             -DUSBCON -DUSBD_USE_CDC -DUSE_USB_HS -DUSE_USB_HS_IN_FS
+lib_extra_dirs             = Marlin/lib
+lib_deps                   = Marlin/lib/rtt
+                             Marlin/lib/fat_fs
+                             Marlin/lib/STM32_USB_Host_Library
```

Comment:
These files are all missing. The reason is probably because the `lib` folder is special and so we filter it out. These things must be moved to the folder named `libs`.